### ISSUE TITLE
Check compatibility with WooPayments

### ISSUE THREAD

**Issue by vikrampm1** [woocommerce/woocommerce-square#121]
The extension doesn’t have a WooPayments compatibility declaration in the readme file. Unlike HPOS and product editing, the WooPayments declaration is not maintained in the extension code. Therefore, it is checked from the readme file.  This work should be considered a low priority as it is expected that standard merchants would use either WooPayments or Square, but unlikely both.  However, it’s not unreasonable that both could be active, so testing and resolving any major incompatibilities would be worthwhile and then over time resolve any lesser issues to avoid poor merchant or shopper experiences who happen to utilize both payment gateway extensions.

**Comment by iamdharmesh** [comment]
### Smoke testing:
- [x] Can activate the plugin without any error
- [x] Verify WooCommerce Square Integration for SSL Compliance
- [x] Check Square extension related tabs and settings should appear
- [x] Connect a Square account
- [x] Check the connection with Square Live/Sandbox account
- [x] Enable/Disable Payment Gateway in WooCommerce Settings
- [x] Verify the functionality of the "Authorization" transaction type
- [x] Verify the functionality of the "Charge Virtual-Only Orders" transaction type
- [x] Verify the functionality of the "Capture Paid Orders" transaction type
- [x] Verify the function "Accepted card logos"
- [x] Verify Customer Profile
- [x] Verify the Digital Wallet settings (Except ApplePay)
- [x] Verify the Digital Wallet Button type and color settings (Except ApplePay)
- [x] Verify the successful import of products from Square to WooCommerce
- [x] Verify the Sync With "Square" and "WooCommerce" SOR
- [x] Verify the Sync With "WooCommerce" SOR
- [x] Verify the Sync With "Square" SOR
- [x] Verify the successful refund process for orders paid through the Square
- [x] Verify the Square Credit Card fields in the Checkout Form
- [x] Verify the add and delete payment method
- [x] Verify the save card
- [x] Verify the Gift card functionality and settings
- [x] Sell/Purchase a giftcard (**WooPayments express Pay Buttons are appear on the product page but it is fixed in #142)**
- [x] Add existing value to giftcard
- [x] Split the payment between Gift card and Credit card (**Non-supported payment methods were showing but it is fixed in #142 -This is not related to WooPayments**)
- [x] Email notification for the giftcard
- [x] Verify checkout using WooCommerce Blocks
- [x] Enable Cash app pay and settings should appear
- [x] Verify Title and Description of Cash App Pay
- [x] Verify Cash App Pay button Theme
- [x] Verify Cash App Pay button Shape
- [x] Verify Payment with Cash App - Desktop
- [x] Verify Payment with Cash App - Mobile
- [x] Verify Cash App Payment Method Availability Based on Address
- [x] Verify Cash App Payment Refund
- [x] Verify Cash app > Transaction Type > Authorization
- [x] Verify Cash app > Transaction Type > Charge Virtual-Only Orders
- [x] Verify Cash app > Transaction Type > Capture Paid Orders
- [x] Cash app > Transaction Type > Charge
- [x] Disable the gateway if the currency is not supported
- [x] Buying a gift card product type using Cash App Pay

### Payments Specific tests:
- Express pay buttons appear on the product page.
    - [x] Google Pay
    - [ ] Apple Pay
    - [ ] Woo Pay

- Express pay buttons appear on the Cart page.
    - [x] Google Pay
    - [ ] Apple Pay
    - [ ] Woo Pay

- Express pay buttons appear on the Checkout page.
    - [x] Google Pay
    - [ ] Apple Pay
    - [ ] Woo Pay

- [x] Payment using credit card and saved credit card

The Square gift card product only supports payment methods added by the Square extension. Therefore, express pay buttons and payment methods added by WooPayments should not be displayed during the checkout process. Currently, express payment buttons appear, but this issue is fixed by PR #142.

While checking on this I found 2 non-related bugs which I have reported separately #140 and #141

@dkotter @peterwilsoncc I am not able to test Apple Pay as it does not appear for me due to geo-location limitations and similarly, I am also not able to complete payment using WooPay (blocked by mobile number and OTP needed - mobile number from India not supported). Could you please suggest the next steps for it?


Thank you.

**Comment by peterwilsoncc** [comment]
I tested the Apple Pay features on a branch with both #124 and #138 merged to ensure Apple Pay displayed as expected and cancelling the payment forms unlocked the forms.

Product page tests:
* [x] Apple Pay (Sq)
* [x] Apple Pay (WooPayments)
* [x] WooPay

Cart page tests 
* [x] Apple Pay (Sq)
* [x] Apple Pay (WooPayments)
* [x] WooPay

Cart page tests (shortcode)
* [x] Apple Pay (Sq)
* [x] Apple Pay (WooPayments)
* [x] WooPay


Checkout page tests
* [x] Apple Pay (Sq)
* [x] Apple Pay (WooPayments)
* [x] WooPay

Checkout page tests (shortcode)
* [x] Apple Pay (Sq)
* [x] Apple Pay (WooPayments)
* [x] WooPay

**Comment by peterwilsoncc** [comment]
@iamdharmesh I've tested both Apple Pay and WooPay buttons and it all appears to be working as expected.

When testing Apple Pay I used a local environment set up with a live WooPayments account and was able to test with both the Square inserted button and the WooPayments inserted button.

For the WooPay button I was able to test on the instawp site.

**Comment by iamdharmesh** [comment]
Thanks a lot for the help with testing this @peterwilsoncc.
